### PR TITLE
Allow the EIP-1820 transaction over RPC unconditionally

### PR DESCRIPTION
**Pull request by schmir-at-bb**
The transaction doesn't use EIP-155 by design and it's safe to allow
it, see https://eips.ethereum.org/EIPS/eip-1820

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1602,14 +1602,20 @@ func (args *SendTxArgs) toTransaction() *types.Transaction {
 	return types.NewTx(data)
 }
 
+// isEIP1820Transaction checks if the given transaction is the EIP1820 transaction. This is allowed
+// to be submitted even if it is not using EIP-155
```
Comment (holiman) [inline internal/ethapi/api.go:1606]: Please add a link to the [EIP](https://eips.ethereum.org/EIPS/eip-1820). To a casual reader, without reading the eip itself, this looks extremely strange.

**Review by holiman — COMMENTED** [review]
Generally looks good to me. It's an interesting case, particularly if we (the Ethereum ecosystem) at one point plans to deprecate non-replay-protected txs, as it is a data-point _against_ doing so. 

Question: how much of a hassle would it be if that happened - would it be just a matter of updating the EIP to use some other deployment method? Presumably, all 'live' chains already have this registry in place.

File: internal/ethapi/api.go (hunk 1 above)
Comment (schmir-at-bb) [reply internal/ethapi/api.go]: I've added link and a bit more information.

**Comment by schmir-at-bb** [comment]
> Question: how much of a hassle would it be if that happened - would it be just a matter of updating the EIP to use some other deployment method? Presumably, all 'live' chains already have this registry in place.

The point of using this deployment method is to end up with the same contract address on any chain. I'm not sure if it's possible to deploy to the same address with an EIP-155 compatible transaction.

**Comment by karalabe** [comment]
I'm unsure about this change really. It adds a burden to every tx send ever on any node. As long as the tx propagation is not limited to only EIP-155, you can always disable this flag for teh time of deployment.

**Comment by schmir-at-bb** [comment]
> I'm unsure about this change really. It adds a burden to every tx send ever on any node. As long as the tx propagation is not limited to only EIP-155, you can always disable this flag for teh time of deployment.

I do not understand your objection. The additional call to isEIP1820Transaction is only done, when the transaction is not using EIP-155 and when the call to SubmitTransaction would return an error anyway. That should certainly not be the case for 'every tx send ever  on any node'.

**Comment by karalabe** [comment]
Because it is a hack that's not needed. I don't see a reason to have it.

**Comment by schmir-at-bb** [comment]
> Because it is a hack that's not needed. I don't see a reason to have it.

https://blog.ethereum.org/2021/03/03/geth-v1-10-0/ mentions that the `-rpc.allow-unprotected-txs` switch is only temporary, which is why I don't want to rely on that and opened the PR.

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -1602,14 +1602,23 @@ func (args *SendTxArgs) toTransaction() *types.Transaction {
 	return types.NewTx(data)
 }
 
+// isEIP1820Transaction checks if the given transaction is the EIP1820 transaction. EIP-1820 is
```
Comment (lightclient) [inline internal/ethapi/api.go:1605]:
```suggestion
// isEIP1820Transaction checks if the given transaction is the EIP-1820 transaction. EIP-1820 is
```

**Comment by holiman** [comment]
I agree with @karalabe about this being supefluous. All it takes is one single node which allows it, in order to broadcast the tx in question. Geth still accepts and relays such transactions

**Comment by schmir-at-bb** [comment]
> All it takes is one single node which allows it, in order to broadcast the tx in question. Geth still accepts and relays such transactions

There is no such node, as I'm running geth in `--dev` mode.

**Comment by MariusVanDerWijden** [comment]
If you're running in `--dev` mode, you can always allow non-eip155 transactions using `--rpc.allow-unprotected-txs`

**Pull request closed without merging**